x/btcstaking/keeper/finality_providers.go

```diff
@@ -67,6 +68,35 @@ func (k Keeper) SlashFinalityProvider(ctx context.Context, fpBTCPK []byte) error
 	return nil
 }
 
+func (k Keeper) PropagateFPSlashingToConsumers(ctx context.Context, fpBTCPK *bbn.BIP340PubKey) error {
+	// Get all delegations for this finality provider
+	delegations, err := k.getFPBTCDelegations(ctx, fpBTCPK)
+	if err != nil {
+		return err
+	}
+
+	for _, delegation := range delegations {
+		// Create SlashedBTCDelegation event
+		consumerEvent := types.CreateSlashedBTCDelegationEvent(delegation)
```

Comment:
This function does not seem optimal indeed.

How about we refactor this parts as following:

- Have a function that given a slashed FP, outputs a map where key is consumer ID and value is a list of `SlashedBTCDelegation` events.
- Given this map, for each consumer ID, we append them to the IBC packet using `AddBTCStakingConsumerEvent`

Here the function can be implemented as follows: for each delegation, gets all involved FPs and then involved consumer IDs, and appends the events to involved consumer IDs in the map